test/libsolidity/SemanticTest.cpp

```diff
@@ -453,6 +453,8 @@ TestCase::TestResult SemanticTest::runTest(
 			test.setFailure(!m_transactionSuccessful);
 			test.setRawBytes(move(output));
 			test.setContractABI(m_compiler.contractABI(m_compiler.lastContractName(m_sources.mainSourceFile)));
+			if (!test.valid())
```

Comment:
This also calls `matchesExpectation()` - what are the consequences of this? Why was it not there before?

Reply:
Thats true, I think it should work just by calling `validateFractionDigits`.

Reply:
I removed `valid` and `validateFractionDigits`.